tests/SettingsBrokerTestDefs.js

```diff
@@ -0,0 +1,231 @@
+/*
+ * GPII App Integration Test Definitions
+ *
+ * Copyright 2017 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * The research leading to these results has received funding from the European Union's
+ * Seventh Framework Programme (FP7/2007-2013)
+ * under grant agreement no. 289016.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    jqUnit = fluid.require("node-jqunit", require, "jqUnit"),
+    gpii = fluid.registerNamespace("gpii");
+
+require("../node_modules/kettle/lib/test/KettleTestUtils.http.js");
+require("../src/main/app.js");
+
+fluid.registerNamespace("gpii.tests.settingsBroker.testDefs");
+
+var liveSettingChange = {
+    path: "textfieldPath",
+    oldValue: "Old value",
+    value: "New Value",
+
+    icon: "../../../../icons/gear-cloud-white.png",
```

Comment:
As with the icon defs above, can we move these towards the package-relative form e.g. "%gpii-app/icons/gear-cloud-white.png"

Reply:
The exact value of the `icon` property (and whether it points to a file which exists or not) does not matter here. For the purpose of the tests it is important that each setting has an `icon` property and that the `src` of its image is the same as the value of the property. As the `fluid.module.resolvePath` does not work properly in this version of the UI tests, I am thinking of not modifying these values for now.

Reply:
Can you say why `fluid.module.resolvePath` does not work properly?

Reply:
Actually, it seems that in the PSP integration tests `fluid.module.resolvePath` works fine. I wanted to use this function in the PSP UI tests, as well, and there it was not working properly so I didn't change the paths that appeared there.